src/practice1/task1/transport/WheeledTransport.java

```diff
@@ -19,10 +34,24 @@ public double getFuelConsumption() {
         return fuelConsumption;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof WheeledTransport)) return false;
+        if (!super.equals(o)) return false;
+        WheeledTransport that = (WheeledTransport) o;
+        return numberOfWheels == that.numberOfWheels && Double.compare(that.fuelConsumption, fuelConsumption) == 0;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), numberOfWheels, fuelConsumption);
```

Comment:
Equals and hashcode shouldn't be based on different fields

Reply:
equals has -> if (!super.equals(o)) return false;

Reply:
equals has --> if (!super.equals(o)) return false;